cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -1370,19 +1371,36 @@ public String findBash() {
 
     String bash = BASH;
     if (SystemInfoImpl.INSTANCE.isWindows()) {
-      bash = findBashOnWindows();
+      String variable = IdeVariables.BASH_PATH.getName();
+      bash = getVariables().get(variable);
+
+      if (bash != null) {
+        Path bashPathVariable = Path.of(bash);
+        if (Files.exists(bashPathVariable)) {
+          debug("{} variable was found and points to: {}", IdeVariables.BASH_PATH, bashPathVariable);
+        } else {
+          warning("{} variable was found at: {} but is not pointing to an existing file", IdeVariables.BASH_PATH, bashPathVariable);
+          bash = null;
+        }
+      } else {
+        debug("{} variable was not found", IdeVariables.BASH_PATH);
+      }
+
       if (bash == null) {
-        String variable = IdeVariables.BASH_PATH.getName();
-        bash = getVariables().get(variable);
+        bash = findBashOnWindows();
         if (bash == null) {
           trace("Bash not found. Trying to search on system PATH.");
           variable = IdeVariables.PATH.getName();
-          Path plainBash = Path.of(BASH);
-          Path bashPath = getPath().findBinary(plainBash);
-          bash = bashPath.toAbsolutePath().toString();
-          if (bash.contains("AppData\\Local\\Microsoft\\WindowsApps")) {
-            warning("Only found windows fake bash that is not usable!");
-            bash = null;
+          if (variable != null) {
+            Path plainBash = Path.of(BASH);
+            Path bashPath = getPath().findBinary(plainBash);
+            bash = bashPath.toAbsolutePath().toString();
+            if (bash.contains("AppData\\Local\\Microsoft\\WindowsApps")) {
+              warning("Only found windows fake bash that is not usable!");
+              bash = null;
+            }
```

Comment:
just to make out dependencies explicit:
Here we need to make use of PR #1592 and convert the `if` into a lambda filter expression including also the `"C:\Windows\System32\bash.exe"` exclusion.

I would not merge this PR unless this is included. So we have to complete and merge PR #1592 first.